scripts/ci/preflight-steps/check-outdated.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/python3
+# Copyright Kani Contributors
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+import subprocess
+
+def run_cargo_command(command, working_directory):
+
+    try:
+        process = subprocess.Popen(
+            command,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            shell=True,
+            text=True,
+            cwd=working_directory)
+        stdout, stderr = process.communicate()
+
+        if process.returncode != 0:
+            print(f"Command '{command}' executed unsuccesfully, in '{working_directory}'.")
+            print(stderr)
+        else:
+            print(f"Command '{command}' executed succesfully, in '{working_directory}'.")
+            print(stdout)
+    except Exception as e:
```

Comment:
Not sure that there's an need for try-except here. If `Popen` raises, I think the program should crash out early.

There's also no need to use `shell=True` here, please pass the command as a list.

More generally I'm not sure that there's any need to buffer the output and then immediately print it. I think this code could be a lot shorter:

```python
cmds = [
  ["cargo", "install", "cargo-outdated"],
  ["cargo", "outdated", "--workspace"],
]
exit_code = 0
for cmd in cmds:
    proc = subprocess.run(cmd, cwd=working_directory)
    exit_code += proc.returncode
sys.exit(exit_code)
```